openapi/external.yaml

```diff
@@ -0,0 +1,164 @@
+        '500':
+          description: Internal Server Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ProblemJson'
+components:
+  securitySchemes:
+    ApiKeyAuth:
+      type: apiKey
+      in: header
+      name: Ocp-Apim-Subscription-Key
+    LollipopAuth:
+      type: apiKey
+      in: header
+      description: A lollipop's JWT auth custom header as `Bearer <JWT Token>`
+      name: x-pagopa-lollipop-auth
+  schemas:
+    ProblemJson:
+      type: object
+      properties:
+        type:
+          type: string
+          format: uri
+          description: |-
+            An absolute URI that identifies the problem type. When dereferenced,
+            it SHOULD provide human-readable documentation for the problem type
+            (e.g., using HTML).
+          default: about:blank
+          example: https://example.com/problem/constraint-violation
+        title:
+          type: string
+          description: >-
+            A short, summary of the problem type. Written in english and
+            readable
+
+            for engineers (usually not suited for non technical stakeholders and
+
+            not localized); example: Service Unavailable
+        status:
+          type: integer
+          format: int32
+          description: >-
+            The HTTP status code generated by the origin server for this
+            occurrence
+
+            of the problem.
+          minimum: 100
+          maximum: 600
+          exclusiveMaximum: true
+          example: 200
+        detail:
+          type: string
+          description: |-
+            A human readable explanation specific to this occurrence of the
+            problem.
+          example: There was an error processing the request
+        instance:
+          type: string
+          format: uri
+          description: >-
+            An absolute URI that identifies the specific occurrence of the
+            problem.
+
+            It may or may not yield further information if dereferenced.
+    AssertionType:
+      type: string
+      enum:
+        - SAML
+        - OIDC
+    AssertionRefSha256:
+      type: string
+      pattern: ^(sha256-[A-Za-z0-9-_=]{1,44})$
+    AssertionRefSha384:
+      type: string
+      pattern: ^(sha384-[A-Za-z0-9-_=]{1,66})$
+    AssertionRefSha512:
+      type: string
+      pattern: ^(sha512-[A-Za-z0-9-_=]{1,88})$
+    AssertionRef:
+      oneOf:
+        - $ref: '#/components/schemas/AssertionRefSha256'
+        - $ref: '#/components/schemas/AssertionRefSha384'
+        - $ref: '#/components/schemas/AssertionRefSha512'
+    OperationId:
+      type: string
+      minLength: 1
+    SamlUserInfo:
+      type: object
+      properties:
+        response_xml:
+          type: string
+          description: A string representation of a signed SPID/CIE response
+          minLength: 1
+      required:
+        - response_xml
+    OidcSignedJwt:
+      type: string
```

Comment:
Just an opinion. Can we create a regex with valid JWT characters to validate JWTs? I know there's not a path traversal issue here, but imho strictly validating the strings we receive from the outside could be a good improvement to never receive anything different from what we expect. In this case a regex could be something like this `^[a-zA-Z0-9-_]+\.[a-zA-Z0-9-_]+\.[a-zA-Z0-9-_]+$`.

Reply:
In this case we have a problem with regex because auto-generated types use javascript PatternString that doesn't accept `\` so we have to escape with double `\` i.e: `^[a-zA-Z0-9-_]+\\.[a-zA-Z0-9-_]+\\.[a-zA-Z0-9-_]+$`

Reply:
If the regex works as expected and you think we can do this, it's ok. It's not a blocker anyway. The risk is non-existent, it's just a fussiness.